src/Bicep.Core/Registry/Oci/ArtifactIdParts.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Bicep.Core.Registry.Oci
+{
+    public class ArtifactIdParts : IArtifactIdParts
```

Comment:
I think the name here is misleading. Consider for example the counter example of a TemplateSpec, although a valid artifact its parts are not composed of a Registry, Repository, Tag &/or Digest. I suggest you name this class `OciArtifactAddressComponents`

Reply:
I like that, thanks.  Had the same issue but couldn't think of anything better.

Reply:
9b757d463